types/validator_set.go

```diff
@@ -349,40 +344,27 @@ func (vals *ValidatorSet) VerifyFutureCommit(newSet *ValidatorSet, chainID strin
 	// Check old voting power.
 	oldVotingPower := int64(0)
 	seen := map[int]bool{}
-	round := commit.Round()
 
 	for idx, precommit := range commit.Precommits {
 		if precommit == nil {
 			continue
 		}
-		if precommit.Height != height {
```

Comment:
shouldn't we still check this once for `commit`?

Reply:
Here we don't check the commit height/blockID in the function, but it does make a call to `VerifyCommit` with the appropriate height/blockID, so it's doing the check there.

https://github.com/tendermint/tendermint/blob/e5965ababe944d7470b4c21c52d09af93a8640b6/types/validator_set.go#L338-L343